app/src/main/java/com/futurice/freesound/feature/search/SearchFragment.kt

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2016 Futurice GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.futurice.freesound.feature.search
+
+import android.os.Bundle
+import android.support.v7.widget.LinearLayoutManager
+import android.view.LayoutInflater
+import android.view.View
+import android.view.View.*
+import android.view.ViewGroup
+import com.futurice.freesound.R
+import com.futurice.freesound.common.rx.plusAssign
+import com.futurice.freesound.core.BindingBaseFragment
+import com.futurice.freesound.feature.common.DisplayableItem
+import com.futurice.freesound.feature.common.scheduling.SchedulerProvider
+import com.futurice.freesound.feature.common.ui.adapter.RecyclerViewAdapter
+import com.futurice.freesound.inject.fragment.BaseFragmentModule
+import com.futurice.freesound.network.api.model.Sound
+import com.futurice.freesound.viewmodel.DataBinder
+import com.futurice.freesound.viewmodel.ViewModel
+import io.reactivex.disposables.CompositeDisposable
+import kotlinx.android.synthetic.main.fragment_search.*
+import timber.log.Timber
+import javax.inject.Inject
+
+class SearchFragment : BindingBaseFragment<SearchFragmentComponent>() {
+
+    @Inject
+    internal lateinit var searchFragmentViewModel: SearchFragmentViewModel
+
+    @Inject
+    internal lateinit var searchResultAdapter: RecyclerViewAdapter<Sound>
+
+    @Inject
+    internal lateinit var schedulerProvider: SchedulerProvider
+
+    private val dataBinder = object : DataBinder {
+
+        override fun bind(disposables: CompositeDisposable) {
+            disposables += searchFragmentViewModel.soundsOnceAndStream
+                    .subscribeOn(schedulerProvider.computation())
+                    .observeOn(schedulerProvider.ui())
+                    .subscribe({ handleResults(it.orDefault { null }) })
```

Comment:
Do you have a bigger picture in mind which involves not exposing `Option` to too many parts of the code? Why not just push down the `Option` to the `handleResults` method as was done previously?

Reply:
I think we should slowly remove my options and replace those with yours. Here, the handleResults logic is a bit simpler when using the null instead of option